src/GCRealTimeMon/Utilities/SpectreConsoleOut.cs

```diff
@@ -0,0 +1,128 @@
+﻿namespace realmon.Utilities
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Diagnostics.Tracing.Analysis.GC;
+    using realmon.Configuration;
+    using realmon.Configuration.Theme;
+    using Spectre.Console;
+
+    /// <summary>
+    /// Utility for writing formatted console output
+    /// </summary>
+    internal class SpectreConsoleOut : IConsoleOut
+    {
+        LiveOutputTable liveOutputTable;
+
+        public SpectreConsoleOut(Configuration configuration)
+        {
+            if (configuration is null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            liveOutputTable = new LiveOutputTable(configuration);
+        }
+
+        public void WriteProcessInfo(string processName, int pid)
+        {
+            WriteRule($"[{ThemeConfig.Current.MessageColor}]Monitoring process with name: [{ThemeConfig.Current.HighlightColor}]{processName}[/] and pid: [{ThemeConfig.Current.HighlightColor}]{pid}[/][/]");
+        }
+
+        public void WriteTableHeaders() => liveOutputTable.Start();
+
+        public void WriteRow(TraceGC gc) => liveOutputTable.WriteRow(gc);
+
+        /// <summary>
+        /// Writes a horizontal rule line with a message in the center to the console.
+        /// </summary>
+        /// <param name="ruleMessage">The message to put in the center of the rule line.</param>
+        public void WriteRule(string ruleMessage)
+        {
+            AnsiConsole.Write(new Rule(ruleMessage).RuleStyle(Style.Parse(ThemeConfig.Current.MessageRuleColor)));
+        }
+
+        /// <summary>
+        /// Writes a horizontal rule line to the console.
+        /// </summary>
+        public void WriteRule()
```

Comment:
what's the definition of a rule?

Reply:
It's a horizontal rule, like the `<hr/>` tag in html.

Reply:
got it, thanks!